### ISSUE TITLE
E208 tests include a task that would raise an Ansible error

### ISSUE THREAD

**Issue by moreati** [ansible/ansible-lint#1007]
##### Summary
In the unit tests for E208 'File permissions not mentioned', there is a task would fail to execute if in an Ansible playbook.

https://github.com/ansible/ansible-lint/blob/cc23bae3c9912aa75658bd3aee75648e80d41dba/test/TestMissingFilePermissionsRule.py#L29-L32

Similarly the documentation for E208 suggests using `mode: preserve`, but a task using that with the file module and `state: file`, or `state: preserve` would similarly fail.

I'm unsure at the moment if the Ansible error is a bug in Ansible or working as intended. I'm raising this in case other lost souls make the change motivated on ansible-lint.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
:~/venvs$ ./ansible-2.9.12/bin/ansible --version
ansible 2.9.12
  config file = /home/alex/.ansible.cfg
  configured module search path = ['/home/alex/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/alex/venvs/ansible-2.9.12/lib/python3.8/site-packages/ansible
  executable location = ./ansible-2.9.12/bin/ansible
  python version = 3.8.2 (default, Jul 16 2020, 14:00:26) [GCC 9.3.0]

:~/venvs$ ./ansible-2.9.12/bin/ansible-lint --version
ansible-lint 4.3.3.dev1+gcc23bae

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
Ubuntu 20.04 x86_64

##### STEPS TO REPRODUCE

```console (paste below)
:~/venvs$ cat playbook.yml 
- name: Test E208
  hosts: localhost
  tasks:
    - name: File module without mode
      file:
        path: canary
        state: file
        mode: preserve

:~/venvs$ touch canary

:~/venvs$ ./ansible-2.9.12/bin/ansible-playbook playbook.yml 
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost
does not match 'all'

PLAY [Test E208] ****************************************************************************************

TASK [Gathering Facts] **********************************************************************************
ok: [localhost]

TASK [File module without mode] *************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "details": "bad symbolic permission for mode: preserve", "gid": 1000, "group": "alex", "mode": "0664", "msg": "mode must be in octal or symbolic form", "owner": "alex", "path": "canary", "size": 0, "state": "file", "uid": 1000}

PLAY RECAP **********************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

##### Desired Behaviour

ansible-lint and ansible should be consistent about whether

```yaml
file:
  path: foo
  mode: preserve
  state: directory

file:
  path: bar
  mode: preserve
  state: file
```

are supported/best practice.

##### Actual Behaviour

ansible-lint suggests/recommends `mode: preserve` in cases that cause an error in Ansible.

**Comment by moreati** [comment]
Same error occurs running the playbook with Ansible 2.10.0a9

**Comment by Sacrome** [comment]
I found that `mode: preserve` is only allowed for the modules `copy` and `template`, contrary to what the documentation states. Other modules like `file` do not have this option built in. 

I've opened a PR on `ansible/ansible` for fixing the documentation at https://github.com/ansible/ansible/pull/71486

**Comment by ssbarnea** [comment]
@Sacrome @moreati Are you aware of any wrong matches still in the linter logic or we can close this one?

**Comment by moreati** [comment]
ansible-lint/test/TestMissingFilePermissionsRule.py (at 0d4142ca5d0921f34d6d53ef01249dcffd776691) still contains the invalid task as a test case. I'll do a PR

**Comment by ssbarnea** [comment]
@moreati @Sacrome please review https://github.com/ansible/ansible-lint/pull/1030 and let me know if it does anything wrong.
